executor/src/witgen/rows.rs

```diff
@@ -260,19 +268,24 @@ impl<T: FieldElement> Row<T> {
 
 impl<T: FieldElement> Row<T> {
     /// Creates a "fresh" row, i.e., one that is empty but initialized with the global range constraints.
-    pub fn fresh(fixed_data: &FixedData<'_, T>, row: RowIndex) -> Row<T> {
+    pub fn fresh<E>(
+        fixed_data: &FixedData<'_, T>,
+        row: impl TryInto<DegreeType, Error = E>,
```

Comment:
Previously, we used `RowIndex`, just to call `RowIndex::into<DegreeType>()`, which essentially returns the given index modulo the number of rows. Now, we can pass e.g. a `usize`, which is sufficient as long as it is guaranteed that `0 <= index < degree`.